src/ima.js

```diff
@@ -1061,7 +1069,7 @@ class Ima extends BasePlugin implements IMiddlewareProvider, IAdsControllerProvi
     if (this._adsLoader) {
       this._adsLoader.contentComplete();
     }
-    if (this._adsManager && this._adsManager.getCuePoints().includes(-1)) {
+    if (this._hasPostRoll()) {
```

Comment:
if post-roll failed _hasPostRoll return true and it won't resolve, I think you need to add the adError check here as well

Reply:
generally, it's true, but in edge case (when post-roll not preloaded) `_adError` could be `true` from mid-roll failure.
anyway, post-roll failure is well handled by the ERROR event so I prefer to not change it.